lib/ABI/FunctionType.cpp

```diff
@@ -740,28 +805,60 @@ Layout::Layout(const model::CABIFunctionType &Function) :
   })) {
 }
 
-Layout::Layout(const model::RawFunctionType &Function) {
-  // Lay register arguments out.
-  for (const model::NamedTypedRegister &Register : Function.Arguments)
-    Arguments.emplace_back().Registers = { Register.Location };
-
+static void
+fillLayoutFromRawFunctionTypeHelper(const model::RawFunctionType &Function,
+                                    abi::FunctionType::Layout &Output) {
   // Lay stack arguments out.
   if (Function.StackArgumentsType.isValid()) {
     const model::Type *OriginalStackType = Function.StackArgumentsType.get();
     auto *StackStruct = llvm::dyn_cast<model::StructType>(OriginalStackType);
     revng_assert(StackStruct,
                  "`RawFunctionType::StackArgumentsType` must be a struct.");
     typename Layout::Argument::StackSpan StackSpan{ 0, StackStruct->Size };
-    Arguments.emplace_back().Stack = std::move(StackSpan);
+    Output.Arguments.emplace_back().Stack = std::move(StackSpan);
   }
 
+  // Fill callee saved registers.
+  Output.CalleeSavedRegisters.resize(Function.PreservedRegisters.size());
+  llvm::copy(Function.PreservedRegisters, Output.CalleeSavedRegisters.begin());
```

Comment:
So this *overwrites* existing entries? `:O`

Reply:
This is just a helper to avoid code duplication. There's never anything in there.

Reply:
No OK, but you can use `llvm::copy` to overwrite entries in a container. I've *always* and only used it with `std::back_inserter`. I just got a new insight into output iterators.

Reply:
Interesting. It'd be nice to have an `append` function that does this in `STLExtras.h`!

Reply:
Is this what you had in mind?
https://github.com/revng/revng/pull/148/commits/093426913074536c0fb649aedcade3be3b18e8d8
https://github.com/revng/revng/pull/148/commits/037706b30087a8df7206bb3ef956f5513f1ef6b5

Reply:
Yes! Maybe add a simple `concept` to catch the presence of `size` and everything else can be easily checked with a concept.

Reply:
Like this?
https://github.com/revng/revng/pull/148/commits/5c87f65c1bb2301931b3425e57eacc7619065eca
https://github.com/revng/revng/pull/148/commits/1b83961229d3e0e2d5a5db3475bc0538935cdaa4

Reply:
Yes, except for `will_be_std` `:D`

Reply:
Any suggestions on how to name that? `:D`

Reply:
I would just put a TODO there.
Several things in `Concepts.h` and `STLExtras.h` are expected to be superseded by standard stuff.
I'm pretty sure we'll review those files once upgrade the compiler or the C++ standard.

So I suggest no namespace, the fact itself that they in one of those two files is a guarantee we'll look into them.

Reply:
https://github.com/revng/revng/pull/148/commits/9acfd377760665ff6fd9163cf2deeaf028c06d3a
https://github.com/revng/revng/pull/148/commits/bc0d874f0ec8f97333080a979a903494e557d179